.github/workflows/maven-build.yml

```
@@ -0,0 +1,34 @@
+name: Maven Build
+
+on:
+  push:
+    branches: [ main, master ]
+  pull_request:
+    types: [ opened, reopened, synchronize, ready_for_review ]
+    branches: [ main, master ]
```

I think this part could be simplified to just any push or pull request. Listing main and master isn't useful as there is no master branch or reason to make one.
```suggestion
on: [push, pull_request]
```